src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -1045,7 +1049,7 @@ protected override void Go()
 
                             while (c-- > 0)
                             {
-                                // Check the timeout every 2000th iteration. The additional if check
+                                // Check the timeout every 2048th iteration. The additional if check
                                 // in every iteration can be neglected as the cost of the CharInClass
                                 // check is many times higher.
                                 if ((uint)c % LoopTimeoutCheckCount == 0)
```

Comment:
Hopefully the JIT can determine that it can just do a bit mask to find N % 2048 and not an expensive mod operation. If not, we could do it explictly if it matters.

Reply:
> Hopefully the JIT can determine that it can just do a bit mask to find N % 2048 and not an expensive mod operation

Yes. Especially that it's unsigned mod.

Reply:
Yes it can or yes it ideally would?

Reply:
Yes it can. It generates an `and` instruction:
```asm
       25FF070000           and      eax, 0x7FF
```